data/datasets/credit_card_default_loader.py

```diff
@@ -98,4 +116,74 @@ def process_data(self, data: pd.DataFrame) -> pd.DataFrame:
                     data[pay_column], pay_mapping
                 )
             data = data.drop(columns=["SEX"])
-        return data
+        return CreditCardDefaultLoader._split_data(data)
+
+    @staticmethod
+    def _split_data(data: pd.DataFrame) -> Mapping[str, pd.DataFrame]:
+        """Randomly splits a DataFrame into train, validation, and test
+        partitions.
+
+        The splits preserve the distribution of labels from the original
+        dataset."""
+        # split the data by label to ensure equal distribution
+        pos_data = data[
+            data[_DATASET_INFO.label_column] == _DATASET_INFO.positive_label
+        ]
+        neg_data = data[
+            data[_DATASET_INFO.label_column] != _DATASET_INFO.positive_label
+        ]
+
+        # get the indices for each split per-label
+        pos_split_indices = CreditCardDefaultLoader._split_indices(
+            pos_data.shape[0],
+            _TRAIN_SPLIT_RATIO,
+            _VAL_SPLIT_RATIO,
+            _TEST_SPLIT_RATIO,
+        )
+
+        neg_split_indices = CreditCardDefaultLoader._split_indices(
+            neg_data.shape[0],
+            _TRAIN_SPLIT_RATIO,
+            _VAL_SPLIT_RATIO,
+            _TEST_SPLIT_RATIO,
+        )
+
+        # construct the splits from the indices
+        data_splits = {}
+        for split in ["train", "val", "test"]:
+            data_splits[split] = pd.concat(
+                [
+                    pos_data.iloc[pos_split_indices[split]],
+                    neg_data.iloc[neg_split_indices[split]],
+                ]
+            )
+
+        return data_splits
+
+    @staticmethod
+    def _split_indices(
+        num_indices: int,
+        train_ratio: float,
+        val_ratio: float,
+        test_ratio: float,
+    ) -> Mapping[str, np.ndarray]:
+        """Partitions a set of indices given the train and validation ratios.
+
+        Returns a mapping from split name ('train', 'val', 'test') to the set
+        of indices that should appear in that split.
+
+        The dataset is assumed to be 0-indexed in increments of 1."""
+        if train_ratio < 0 or val_ratio < 0 or test_ratio < 0:
+            raise RuntimeError("Dataset split ratios must be non-negative.")
+        if np.abs((train_ratio + val_ratio + test_ratio) - 1) > 1e-5:
```

Comment:
Can you please add a comment to make it clear that this is checking that the different ratios sum to approx. 1?